### ISSUE TITLE
How to generate output in same directory structure as src?

### ISSUE THREAD

**Issue by probins** [oscarotero/lume#1]
I'm wanting to convert a simple metalsmith site to run on deno, and Lume looks like a reasonable fit. One thing I'm stuck on is the structure of the output site. I want to convert `src/abc.md` to `site/abc.html` and `src/xyz.md` to `site/xyz.html`, but it puts each src file in its own directory, which I don't want. I tried using the `permalink` variable, but this makes no difference. Seems to be in `site.js` but I can't see any way to define this.

**Comment by oscarotero** [comment]
Hi.
Yes, Lume creates pretty urls, so if you export a page as `/about-us`, the real filename is `/about-us/index.html`. But I understand that this is not always desirable. A quick fix is adding the `.html` extension to the permalink variable (ex: `permalink: about-us.html`). Didn't that work for you?

Anyway, I'm going to create a configuration option to disable permalinks, so you don't have to edit the permalink in every page.

**Comment by oscarotero** [comment]
Done!
The new version `0.7.0` has an option do disable prettyUrls. Just set `false` to this option in the `_config.js` file:

```js
const site = lume({
  prettyUrls: false
})
```

**Comment by probins** [comment]
wow, that was a fast response :-)

Adding the permalink option to the pages makes no difference.

Now I'm having another problem, which is probably due to my limited knowledge of deno. I installed 0.6.0 yesterday with `deno install`. How do I update to the new version? There doesn't seem to be an update or uninstall option, and running install without a release no uses 0.6.0. Using `deno run` without a release no uses 0.6.0, and specifying 0.7.0 loads 0.7.0 `cli.js`, but then uses 0.6.0 for all other files, including `site.js`, which is the one that's changed. 😕

**Comment by probins** [comment]
PS permalink works in 0.6.0 if I use `.htm`. `.html` is treated in the same way as no permalink.

**Comment by oscarotero** [comment]
Yes, I didn't find a good way to manage and update versioning.
I've released `0.7.2` that fixes the problem with permalink.
Updating a package is the same as installing it, but you need to add the `-f` to force the installation (and override the previous installation), and `--reload` to remove the cache and reload all dependencies:

```sh
deno install -f --reload --unstable --allow-read --allow-write --allow-net https://deno.land/x/lume/cli.js
```

Note also that in `_config.js` you can return a lume instance of any version, different to what do you have installed in cli, for example:

```js
import lume from "https://deno.land/x/lume@v0.7.2/mod.js";
  
const site = lume();

export default site;
```

So, other way to update the version is just editing the version in the `_config.js` file.

**Comment by probins** [comment]
yay, that now works. Thanks for the lesson in using Deno. I see you can also use `-r` instead of `--reload`, so a bit less to type.

Moitas grazas (as you name your software with a Galician word, I assume you are galego :-))
